dss/index/indexer.py

```diff
@@ -35,9 +40,9 @@ def process_new_indexable_object(self, event: Mapping[str, Any]) -> None:
     def index_object(self, key):
```

Comment:
There seems to be an (understandable) mixing of indexing logic and timing logic, but I wonder if these concerns could be separated.

For instance,
```
num_ops = indexer.load_data(*args)
check_time(num_ops * indexer.time_for_one_thing)
indexer.do_it()
```

Reply:
To my understanding, @xbrianh is emphasizing where the logic lives. I think separating the concerns is possible by returning a data structure that defers the work once you know the size like the one below.

```python
class Task:

    def __init__(self, size, logic, *args, **kwargs):
        self.size = size
        self.logic = logic
        self.args = args
        self.kwargs = kwargs

    def submit(self):
        result = self.logic(*self.args, **self.kwargs)
        self.size = 0
        return result
```

So you could do something like.

```python
def foo(n):
    for i in range(n):
        time.sleep(1)

def some_insertion_logic():
    ...
    return Task(5, foo, 5)

# in a different (parent) module
print("Unexecuted task size is: " + str(task.size))

if task.size < 10:
    result = task.submit()
```

Just a thought.

Reply:
To summarize our F2F conversation @mweiden, I don't think that the extra complexity of lifting  the unit of work (that's the design pattern your Task class follows) is warranted if we're moving the logic within the class and just one layer up in the call hierarchy.

The thing to keep in mind is that this PR already consolidates timing logic that was previously spread around more.